src/test/scala/pirate.spec/Laws.scala

```diff
@@ -1,205 +1,462 @@
+    } yield {
+      F.traverseLaw.naturality[N, M, A](nat)(fma) ==== true
+    }
 
-        property("purity.option") = purity[F, Option, Int]
-        property("purity.stream") = purity[F, Stream, Int]
+    def parallelFusion[F[_], N[_], M[_], A, B](
+      genFA: Gen[F[A]]
+    , genAMB: Gen[A => M[B]]
+    , genANB: Gen[A => N[B]]
+    )(
+      implicit F: Traverse[F], N: Applicative[N], M: Applicative[M], MN: Equal[(M[F[B]], N[F[B]])]
+    ): Property = for {
+      fa <- genFA.log("fa")
+      amb <- genAMB.log("amb")
+      anb <- genANB.log("anb")
+    } yield {
+      F.traverseLaw.parallelFusion[N, M, A, B](fa, amb, anb) ==== true
+    }
 
-        property("sequential fusion") = sequentialFusion[F, Option, List, Int, Int, Int]
-      }
+    def laws[F[_]](
+      name: String
+    , genFInt: Gen[F[Int]]
+    , genIntToInt: Gen[Int => Int]
+    , genIntToListInt: Gen[Int => List[Int]]
+    , genIntToOptionInt: Gen[Int => Option[Int]]
+    )(
+      implicit F: Traverse[F], EF: Equal[F[Int]]
+    ): List[Test] = {
+      import std.list._, std.option._
+      val lawName = "Traverse Law"
+      List(
+        property(s"$name - $lawName: identity traverse", identityTraverse[F, Int, Int](genFInt, genIntToInt))
+      , property(s"$name - $lawName: purity.option", purity[F, Option, Int](genFInt))
+      , property(s"$name - $lawName: purity.stream", purity[F, Stream, Int](genFInt))
+      , property(s"$name - $lawName: sequential fusion", sequentialFusion[F, Option, List, Int, Int, Int](genFInt, genIntToListInt, genIntToOptionInt))
+      )
+    }
   }
 
 
   object plus {
-    def associative[F[_], X](implicit f: Plus[F], afx: Arbitrary[F[X]], ef: Equal[F[X]]) =
-      forAll(f.plusLaw.associative[X] _)
-
-    def laws[F[_]](implicit F: Plus[F], afx: Arbitrary[F[Int]], ef: Equal[F[Int]]) = new Properties("plus") {
-      include(semigroup.laws[F[Int]](F.semigroup[Int], implicitly, implicitly))
-      property("associative") = associative[F, Int]
+    def associative[F[_], A](genFA: Gen[F[A]])(implicit f: Plus[F], ef: Equal[F[A]]): Property = for {
+      fa1 <- genFA.log("fa1")
+      fa2 <- genFA.log("fa2")
+      fa3 <- genFA.log("fa3")
+    } yield {
+      f.plusLaw.associative[A](fa1, fa2, fa3) ==== true
     }
+
+    def laws[F[_]](name: String, genFInt: Gen[F[Int]])(implicit F: Plus[F], ef: Equal[F[Int]]): List[Test] =
+      semigroup.laws[F[Int]](name, genFInt)(F.semigroup[Int], implicitly) ++
+      List(
+        property(s"$name - Plus Law: associative", associative[F, Int](genFInt))
+      )
   }
 
   object plusEmpty {
-    def leftPlusIdentity[F[_], X](implicit f: PlusEmpty[F], afx: Arbitrary[F[X]], ef: Equal[F[X]]) =
-      forAll(f.plusEmptyLaw.leftPlusIdentity[X] _)
+    def leftPlusIdentity[F[_], A](genFA: Gen[F[A]])(implicit f: PlusEmpty[F], ef: Equal[F[A]]): Property =
+      for {
+        fa <- genFA.log("fa")
+      } yield {
+        f.plusEmptyLaw.leftPlusIdentity[A](fa) ==== true
```

Comment:
Super nit but prefer `Result.assert`

Reply:
OK, I will.

Reply:
Actually don't bother. I really don't care that much. :)

Reply:
Already done. 🤣